src/base/actions/edit-mode-action.ts

```diff
@@ -0,0 +1,35 @@
+/********************************************************************************
+ * Copyright (c) 2020 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { Action } from "sprotty";
+import { injectable } from "inversify";
+
+@injectable()
+export class SetEditModeAction implements Action {
+    static readonly KIND = "setEditMode";
+    kind = SetEditModeAction.KIND;
+
+    constructor(public readonly editMode: string = EditMode.EDITABLE) { }
```

Comment:
I agree it would be nicer, but I opted against it because it  restricts extensibility. We could not introduce a new edit mode type in glps-based projects without adapting the base code.

Reply:
In addition, it's not possible to define the editMode the same way on the server side so we still would need the constant-based approach there

Reply:
Ok, I see...